sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleBackendMode.java

```diff
@@ -0,0 +1,434 @@
+        final int dow = calendar.get(Calendar.DAY_OF_WEEK) - 1;
+
+        request.params.add("dow", dow);
+        request.params.add("hour", hour);
+        request.params.add("timestamp", timestamp);
+        request.params.add("tz", DeviceCore.dev.getTimezoneOffset());
+    }
+
+    private void addEventsAgainstDeviceIdToRequestQ(String deviceID, JSONArray events) {
+        //TODO: Need to verify order of events.
+        Request request = new Request();
+        request.params.add("device_id", deviceID);
+        request.params.add("events", events);
+        addTimeInfoIntoRequest(request, System.currentTimeMillis());
+        request.own(ModuleBackendMode.class);
+        requestQ.add(request);
+    }
+
+    private void addEventsToRequestQ() {
+        for (Map.Entry<String, JSONArray> entry : eventQueues.entrySet()) {
+            addEventsAgainstDeviceIdToRequestQ(entry.getKey(), entry.getValue());
+        }
+        eventQSize = 0;
+        eventQueues.clear();
+
+        processRequestQ();
+    }
+
+    private void processRequestQ() {
+        if (defferUpload) {
+            return;
+        }
+
+        if (tasks.isRunning() || requestQ.size() == 0) {
+            return;
+        }
+
+        Request request = requestQ.remove();
+        tasks.run(sendRequest(request));
+    }
+
+    private Tasks.Task<Boolean> sendRequest(final Request request) {
+        return new Tasks.Task<Boolean>(Tasks.ID_STRICT) {
+            @Override
+            public Boolean call() throws Exception {
+                if (request == null) {
+                    return false;
+                } else {
+                    L.d("Preparing request: " + request);
+                    final Boolean check = SDKCore.instance.isRequestReady(request);
+                    if (check == null) {
+                        L.d("Request is not ready yet: " + request);
+                        return false;
+                    } else if (check.equals(Boolean.FALSE)) {
+                        L.d("Request won't be ready, removing: " + request);
+                        return true;
+                    } else {
+                        tasks.run(transport.send(request), new Tasks.Callback<Boolean>() {
+                            @Override
+                            public void call(Boolean result) throws Exception {
+                                L.d("Request " + request.storageId() + " sent?: " + result);
+                                processRequestQ();
+                            }
+                        });
+                        return true;
+                    }
+                }
+            }
+        };
+    }
+
+    public int getEventQSize() {
+        return eventQSize;
+    }
+
+    public void setEventQSize(int eventQSize) {
+        this.eventQSize = eventQSize;
+    }
+
+    public Queue<Request> getRequestQ() {
+        return requestQ;
+    }
+
+    public Map<String, JSONArray> getEventQueues() {
```

Comment:
No need for this function

Reply:
![image](https://user-images.githubusercontent.com/8924406/158343574-1bc674dc-a4e5-43c5-9eb2-174f1a823d89.png)
These four call are being used in tests classes. otherwise I have to make them public.

Reply:
ok

Reply:
setEventQSize should not be controlled with this but with the init config object. So that should still be removed